app.py

```diff
@@ -101,7 +110,11 @@ def get_schedule(username):
     schedules = get_schedule_data()
     if username not in schedules:
         return None
-    return schedules[username]
+    pfschedule = copy.copy(schedules[username])
+    real_name = get_preferred_name(pfschedule)
+    if real_name != None:
+        pfschedule["firstname"] = get_preferred_name(pfschedule)
```

Comment:
don't need to call `get_preferred_name` twice...
```suggestion
        pfschedule["firstname"] = real_name
```